specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,557 @@
+  /// by the Javascript, or the host's Stop API call.
+  /// TextureBuffer related API calls after this event will return an error
+  /// of HRESULT_FROM_WIN32(ERROR_INVALID_STATE).
+  HRESULT add_StopRequested(
+      [in] ICoreWebView2StagingTextureStreamStopRequestedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove listener for stop stream request.
+  HRESULT remove_StopRequested(
+      [in] EventRegistrationToken token);
+  /// Creates TextureBuffer that will be referenced by the host and the browser.
+  /// By using the TextureBuffer mechanism, the host does not have to
+  /// send respective texture to the renderer, instead it notifies it
+  /// with internal TextureBuffer id, which is the identity of the TextureBuffer.
+  /// The TextureBuffer is 2D texture, IDXGIResource, format and will be
+  /// exposed through shared HANDLE or IUnknown type through ICoreWebView2StagingTexture.
+  /// Whenever the host has new texture to write, it should ask
+  /// reusable ICoreWebView2StagingTexture from the GetAvailableBuffer,
+  /// which returns ICoreWebView2StagingTexture.
+  /// If the GetAvailableBuffer returns an error, then the host calls the
+  /// CreateBuffer to allocate new TextureBuffer.
+  /// The API also registers created shared handle to the browser once it
+  /// created the resource.
+
+  /// Unit for width and height is texture unit (in texels).
+  /// 'https://learn.microsoft.com/en-us/windows/win32/api/d3d11ns-d3d11-d3d11_texture2d_desc'
+  HRESULT CreateBuffer(
+    [in] UINT32 width,
+    [in] UINT32 height,
+    [out, retval] ICoreWebView2StagingTexture** buffer);
+  /// Returns reuseable TextureBuffer for video frame rendering.
+  /// Once the renderer finishes rendering of TextureBuffer's video frame, which
+  /// was requested by Present, the renderer informs the host so that it can
+  /// be reused. The host has to create new TextureBuffer with CreateBuffer
+  /// if the API return an error HRESULT_FROM_WIN32(ERROR_NO_MORE_ITEMS).
+  HRESULT GetAvailableBuffer([out, retval] ICoreWebView2StagingTexture** buffer);
+  /// Remove TextureBuffer when the host removes the backed 2D texture.
+  /// The host can save the existing resources by deleting 2D textures
+  /// when it changes the frame sizes.
+  HRESULT RemoveBuffer([in] ICoreWebView2StagingTexture* buffer);
+  /// Indicates that the TextureBuffer is ready to present.
+  /// The TextureBuffer must be retrieved from the GetAvailableBuffer.
+  /// The host writes new texture to the local shared 2D texture of
+  /// the TextureBuffer id, which is created via CreateBuffer.
+  /// SetBuffer API can be called in any thread.
+
+  /// `timestamp` is video capture time with unit of microseconds units.
+  /// The value does not have to be exact captured time, but it should be
+  /// increasing order, the next Present's TextureBuffer should have later
+  /// time.
+  HRESULT SetBuffer([in] ICoreWebView2StagingTexture* buffer,

```

Comment:
SetBuffer sounds like it is setting a property. Like there is one buffer member inside the texture stream and when you call SetBuffer its changing that one member to point to the new provided parameter. Is that what is happening? If SetBuffer is adding to a list of buffers then it should be called AddBuffer (and the RemoveBuffer method is well named Add/Remove go together generally). If this is SetBuffer, then RemoveBuffer should probably be called ClearBuffer, and it wouldn't need to take a parameter.

Reply:
SetBuffer is single item only so the latest SetBuffer will override the previous ones.

RemoeBuffer is actually removing a TextureBuffer. I will change a name to RemoveTextureBuffer that will remove a buffer from the browser side (The browser received and kept it when the host call Present).